### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by rdaly525**
Major changes:
-New simulator plugin API
-New logging methodology/API
-Moved linebuffer/unified buffer out of commonlib into a separate repo. https://github.com/joyliu37/BufferMapping

### CODE REVIEW

**Comment by rdaly525** [comment]
https://github.com/StanfordAHA/aha/tree/coreir-dev Tracking if this affects aha/GarnetFlow

File: .gitignore [hunk 1]
```diff
@@ -51,3 +51,14 @@ a.out.dSYM/
 
 conv_3_1.o
 include/verilogAST.hpp
+tests/unit/bit_replacement.json
```
Comment (rsetaluri) [inline .gitignore:54]: are these gold files? might be easier to put them into `tests/gold/` and then ignore `tests/gold/*`

File: include/coreir/common/logging_lite.hpp [hunk 2]
```diff
@@ -74,7 +86,7 @@ class LoggerVoidify {
 }  // namespace common
 
 #define LOG(severity)                                                   \
-  ::common::internal::LoggerWrapper(severity == FATAL)                  \
+  ::common::internal::LoggerWrapper(severity)                           \
```
Comment (rsetaluri) [inline include/coreir/common/logging_lite.hpp:89]: nit: whitespace

File: include/coreir/simulator/interpreter.h [hunk 3]
```diff
@@ -120,14 +120,15 @@ namespace CoreIR {
...
+  typedef float (*FloatOp)(const float l, const float r);
+  
   typedef std::function<bool()> StopFunction;
 
   struct StopCondition {
     std::string name;
     StopFunction stopTest;
   };
-
+
```
Comment (rsetaluri) [inline include/coreir/simulator/interpreter.h:131]: nit: whitespace

File: .gitignore (hunk 1 above)
Comment (rdaly525) [reply .gitignore:54]: Added task in CoreIR cleanup to do the build/ gold/ flow for tests.

**Pull request merged**